### ISSUE TITLE
Convert heterogeneous NetworkX graph to HeteroData without normalizing features across node types

### ISSUE THREAD

**Issue by JosephDenman** [pyg-team/pytorch_geometric#7340]
### 🚀 The feature, motivation and pitch

I am working on converting a heterogeneous NetworkX graph to `HeteroData`. Here is an example that I would like to see working:

```python
nodes_types = torch.tensor([1, 2, 3])
nx_graph = nx.MultiGraph()
nx_graph.add_node(0, type=1, foo=4.97, bar=torch.tensor([9, 8]))
nx_graph.add_node(1, type=2, foo=4.97)
nx_graph.add_node(2, type=3)
pyg.utils.from_networkx(nx_graph, group_node_attrs=['type', 'foo', 'bar'])
```
which currently throws `raise ValueError('Not all nodes contain the same attributes')`.

I know that my graph is heterogeneous, but to convert from NetworkX to HeteroData, I need to manually normalize all of the node attributes. I would like to not have to write my own conversion layer.

### Alternatives

N/A

### Additional context

N/A

**Comment by rusty1s** [comment]
As discussed on Slack, this is not possible with the current `from_networkx` interface. The challenge here is that `networkx` is very unrestricted of how a heterogeneous graph can look like, while PyG expects that nodes of the same type hold the same of attributes. One idea for an interface would be that one can specify which attributes in `networkx` denote a node type/edge type, e.g.:
```python
from_hetero_networkx(G, node_type_attribute: str, edge_type_attribute: str)
```

**Comment by JosephDenman** [comment]
Yep, this is just to record the feature request.

**Comment by rusty1s** [comment]
Yes, ofc. I marked it as "P0" and "Help wanted". If anyone has capacity to work on this, please go ahead. Otherwise, I will try to add it.

**Comment by SimonPop** [comment]
Hello @rusty1s , @JosephDenman ,

I'd gladly work on that topic if it is fine with both of you :)

I would start implementing the interface you proposed, and take suggestions from that point onward. That is:

```python
from_hetero_networkx(G, node_type_attribute: str, edge_type_attribute: str)
```

Should any piece of code be similar to the original `from_networkx` method, I'd try to factorize them.

**Comment by JosephDenman** [comment]
Please do!

**Comment by rusty1s** [comment]
Added the reverse direction via `to_networkx` in https://github.com/pyg-team/pytorch_geometric/pull/7713.

**Comment by SimonPop** [comment]
Hello again,

I created a pull request for this issue right there: https://github.com/pyg-team/pytorch_geometric/pull/7744 

It does implement the interface mentioned above and does so in order that `to_networkx` gives the inverse function.
